core/src/main/java/org/elasticsearch/common/blobstore/fs/FsBlobContainer.java

```diff
@@ -85,7 +86,13 @@ public FsBlobContainer(FsBlobStore blobStore, BlobPath blobPath, Path path) {
     @Override
     public void deleteBlob(String blobName) throws IOException {
         Path blobPath = path.resolve(blobName);
-        Files.deleteIfExists(blobPath);
+        try {
+            if (!Files.deleteIfExists(blobPath)) {
+                throw new IOException("File " + blobPath.toString() + " does not exist");
+            }
+        } catch (DirectoryNotEmptyException | SecurityException e) {
+            throw new IOException(e);
```

Comment:
I think it's mistake to wrap a `SecurityException` in an `IOException`, I think that the Javadocs need to changed. And again, a `DirectoryNotEmptyException` is already an `IOException` it doesn't need to wrapped and it allow it to bubble up does not violate what should be the contract for this method.